### PR TITLE
Feature/#169364025/welcome widget with client

### PR DESCRIPTION
**Pull request by andrewchuhlomin**

### CODE REVIEW

File: packages/bootstrap/package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "storybook": "start-storybook -p 6006"
   },
   "dependencies": {
+    "@hiro-graph/client": "^6.0.21",
```
Comment (alex-melnik) [inline packages/bootstrap/package.json]: Could we move it to devDeps?

File: packages/bootstrap/src/components/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
-        )}
-      </DashboardContextProvider>
+      {Object.entries(config.widgets || []).map(
+        ([widgetConfigKey, widgetConfig]) => (
+          <DashboardContextProvider
+            key={widgetConfigKey}
+            client={client}
+            provider={provider}
+            config={widgetConfig}
+            theme={config.theme}
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/Dashboard.tsx]: Does widget need to know about global dashboard theme?

File: packages/bootstrap/src/components/dashboard/context.tsx [hunk 3]
```diff
@@ -1,38 +1,52 @@
...
+  config: WidgetConfig;
 }
 
-const DashboardContext = React.createContext<DashboardDataProvider | {}>({});
+const DashboardContext = React.createContext<DashboardContextValue | {}>({});
 
 // Wrapped access and provider for context
 export const useDashboardContext = () =>
-  useContext(DashboardContext) as DashboardDataProvider;
+  useContext(DashboardContext) as DashboardContextValue;
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/context.tsx]: If we use this context per widget, context looks like `WidgetContext` , what do you think about naming? @andrewchuhlomin

File: packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx [hunk 4]
```diff
@@ -148,7 +211,34 @@ export const Welcome: React.FC<WelcomeProps> = ({ config: { title } }) => {
   return (
     <Box centered>
       {title && <small>{title}</small>}
-      {loading ? <Loader active /> : content}
+      {content}
+      <Loader active={loading} />
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx]: I'd like to recommend to use `React.Suspense` for handling some async flow on UI

File: packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx [hunk 5]
```diff
@@ -68,36 +68,99 @@ const SessionsButtonText: React.FC<SessionsButtonTextProps> = ({
...
+      } catch (error) {
+        setStartedSessionsCount({
+          result: 0,
+          loading: false,
+          error,
+        });
+      }
+    };
+
+    fetchData();
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx]: Fetching methods `startedSessionsCount` and `newSessionsCount` are the same (DRY)

File: packages/bootstrap/src/components/index.spec.tsx [hunk 6]
```diff
@@ -9,62 +10,51 @@ import * as modules from './modules';
...
+    },
+  );
 
-  describe(folderName, () => {
-    Object.entries(components)
-      .filter(([name]) => !name.endsWith('Context'))
-      .forEach((c) => {
-        const [name, Component] = c;
+  beforeAll(() => {
+    const mock = jest.spyOn(client, 'lucene');
```
Comment (alex-melnik) [inline packages/bootstrap/src/components/index.spec.tsx]: 👍

File: packages/bootstrap/src/components/dashboard/context.tsx (hunk 3 above)
Comment (andrewchuhlomin) [reply packages/bootstrap/src/components/dashboard/context.tsx]: @alex-melnik I agree with you

File: packages/bootstrap/src/components/dashboard/widgets/Demo.tsx [hunk 7]
```diff
@@ -1,41 +0,0 @@
-import React from 'react';
```
Comment (yury-klika) [inline packages/bootstrap/src/components/dashboard/widgets/Demo.tsx]: Demo Widget is used for Storybook's Dashboard layout preview

File: packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx (hunk 4 above)
Comment (andrewchuhlomin) [reply packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx]: Unfortunately, we cannot use it now 
https://reactjs.org/docs/concurrent-mode-suspense.html#what-if-i-dont-use-relay

File: packages/bootstrap/src/components/dashboard/widgets/Demo.tsx (hunk 7 above)
Comment (yury-klika) [reply packages/bootstrap/src/components/dashboard/widgets/Demo.tsx]: It's not a big deal, but Dashboard preview becomes a bit more complicated without Demo Widget - we need to be sure that "real" Widgets are properly implemented and mocked to just preview Dashboard's Layout. On the other hand the simple Demo Widget allows to just try different approaches for the layout without any risk to be blocked by not readiness of "real" Widget

File: packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx (hunk 5 above)
Comment (yury-klika) [reply packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx]: Both these `useEffect` hooks are mostly the same - the difference is in just 2 items:

- data fetching function
- state hook

I believe these are good candidates for custom hook

File: packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx [hunk 8]
```diff
@@ -148,7 +211,34 @@ export const Welcome: React.FC<WelcomeProps> = ({ config: { title } }) => {
...
     <Box centered>
       {title && <small>{title}</small>}
-      {loading ? <Loader active /> : content}
+      {content}
+      <Loader active={loading} />
     </Box>
   );
 };
+
+const fetchNewSessionsCount = async (client: Client) => {
```
Comment (yury-klika) [inline packages/bootstrap/src/components/dashboard/widgets/Welcome.tsx]: Same here - these could be replaced with single function with `filterQuery` argument

**Pull request merged**